snuba/query/parser/simple_arithm_expressions.py

```diff
@@ -0,0 +1,50 @@
+from parsimonious.grammar import Grammar
+from parsimonious.nodes import NodeVisitor
+
+from snuba.query.expressions import FunctionCall, Literal
+
+grammar = Grammar(
+    r"""
+    expression = term (("+" expression) / space)
+    term = factor (("*" term) / space)
+    factor = ~"[1-9]\d*"
+    space = ''
+
+"""
+)
+
+# TODO: Expand grammar to include "-" and "/" operators.
+
+
+class PrefixVisitor(NodeVisitor):
+    def visit_space(self, node, children):
+        return
+
+    def visit_factor(self, node, children):
+        return Literal(None, int(node.text))
+
+    def visit_term(self, node, children):
+        factor, term = children
+
+        if all(x is None for x in term):
+            # A check for any None child nodes
+            # that arose from empty space ''
+            return factor
+        else:
+            return FunctionCall(None, term[0][0], (factor, term[0][1]))
```

Comment:
We have some helper functions to build the right FunctionCall structure (that uses the right function name for arithmetic operations)
https://github.com/getsentry/snuba/blob/master/snuba/query/dsl.py#L25

Reply:
New version uses them.